lib/scryptenc/scryptenc.c

```diff
@@ -269,10 +270,10 @@ scryptdec_setup(const uint8_t header[96], uint8_t dk[64],
 	/*
 	 * Check whether the provided parameters are valid and whether the
 	 * key derivation function can be computed within the allowed memory
-	 * and CPU time.
+	 * and CPU time, unless the user chose to disable this test.
 	 */
-	if ((rc = checkparams(maxmem, maxmemfrac, maxtime, logN, r, p,
-	    verbose)) != 0)
+	if (!force_resources && (rc = checkparams(maxmem, maxmemfrac, maxtime,
```

Comment:
We need to check if the parameters are valid even if we're not trying to limit CPU or RAM usage.